localstripe/resources.py

```diff
@@ -1298,7 +1298,8 @@ def _set_up_subscription_and_invoice(self, plan, create_an_invoice=True):
                             description='Unused time',
                             customer=self.customer))
 
-        if create_an_invoice:
+        # Don't charge for a free plan
+        if create_an_invoice and plan.amount != 0:
```

Comment:
I'm not sure it's that simple: I think it's possible invoice have an `amount != 0` if the the plan `amount == 0` (prorata, coupon, refund?)

Reply:
I've made some research (please see bellow) and if I understood well, stripe will never change plan amount value directly.

Because of this, I think this method is safe.
If the plan amount is 0, the plan is a free plan.

Some urls I found related to this in documentation and reference :
* (Plan) https://stripe.com/docs/api/plans/create#create_plan-amount
    *A positive integer in cents (or 0 for a free plan) representing how much to charge on a recurring basis.*
* (Coupon) https://stripe.com/docs/billing/subscriptions/discounts#creating-coupons
    *Coupons have the following parameters:*
    *[...]*
    *percent_off or amount_off*
* (Coupon) https://stripe.com/docs/billing/subscriptions/discounts#discounting-subscriptions
    ```bash
    curl https://api.stripe.com/v1/subscriptions \
       -u sk_test_kFvN9JMxlDgTbtng2oeNYXii: \
       -d customer=cus_4fdAW5ftNQow1a \
       -d items[0][plan]=plan_CBb6IXqvTLXp3f \
       -d coupon=free-period
    ```
* (Prorations) https://stripe.com/docs/billing/subscriptions/prorations#previewing-proration
    A proration_date have to be passed while modifying subscription or prorate=True but the plan  amount will not be changed directly.
* (Refund) https://stripe.com/docs/refunds#api

Reply:
I've just checked on my Stripe account, and users subscribed to free plans **do have invoices**. Their invoices mention "€0.00", but they exist and are issued every month.

So maybe a better way to achieve this would be to prevent creation of charge if amount is zero? I'm thinking about:
https://github.com/adrienverge/localstripe/blob/0d42e0af9ebe0f3cae4708a0545f72b55d5f6855/localstripe/resources.py#L682

Reply:
Woo, sorry, I read your previous comment completely wrong. I don't know why but I was focussed on stripe allowing plan amount to be zero or not...

I will try to give it a closer look soon, sory again for the miscomprehension.

Reply:
No problem!